sql/parser/builtins.go

```diff
@@ -851,24 +851,24 @@ var substringImpls = []builtin{
 		types:      typeList{stringType, intType},
 		returnType: DummyString,
 		fn: func(_ EvalContext, args DTuple) (Datum, error) {
-			str := args[0].(DString)
+			runes := []rune(string(args[0].(DString)))
```

Comment:
Yeah, the `string()` is needed.

Agreed, this is inefficient, but this isn't the only place. More discussion on this in #3075.